### PR TITLE
Implemented Relationships Table & Adding / Removing Friends

### PR DESCRIPTION
**Pull request by cmyui**

### CODE REVIEW

File: server/geolocation.py [hunk 1]
```diff
@@ -38,4 +38,7 @@
 
 # TODO: more formal/correct naming
 def country_str_to_int(value: str) -> int:
+    if not len(value) == 2:
+        return
+
```
Comment (cmyui) [inline server/geolocation.py]: we don't want to do this atm
```suggestion
```

File: server/main.py [hunk 2]
```diff
@@ -248,10 +250,12 @@ async def handle_login(request: Request) -> Response:
     # osu chat channels
     for channel in await channels.fetch_all():
         # TODO: privilege check - do they have access to this channel?
+        current_channel_members = await channel_members.members(channel["channel_id"])
+        
         response_data += packets.write_channel_info_packet(
             channel["name"],
             channel["topic"],
-            channel["num_sessions"],
+            len(current_channel_members)
```
Comment (cmyui) [inline server/main.py:258]: nice fix

File: server/packet_handlers.py [hunk 3]
```diff
@@ -188,6 +192,59 @@ async def request_status_update_handler(session: "Session", packet_data: bytes):
...
+    packet_reader = packets.PacketReader(packet_data)
+    channel_name = packet_reader.read_string()
+
+    channel = await channels.fetch_one_by_name(channel_name)
+
+    if not channel:
+        return
+
+    if session["session_id"] in await channel_members.members(channel["channel_id"]):
+        return
```
Comment (cmyui) [inline server/packet_handlers.py]: would like these operations separately for clarity
```suggestion
    channel_members = await channel_members.members(channel["channel_id"])

    if session["session_id"] in channel_members :
        return
```

File: server/packet_handlers.py [hunk 4]
```diff
@@ -188,6 +192,59 @@ async def request_status_update_handler(session: "Session", packet_data: bytes):
...
+
+    await channel_members.add(channel["channel_id"], session["session_id"])
+
+
+@bancho_handler(packets.ClientPackets.FRIEND_ADD)
+async def user_adds_friend_handler(session: "Session", packet_data: bytes):
+    packet_reader = packets.PacketReader(packet_data)
+    user_being_friended_id = packet_reader.read_i32()
+
+    await relationships.create(session["account_id"], user_being_friended_id, "friend")
```
Comment (cmyui) [inline server/packet_handlers.py]:
```suggestion
    target_id = packet_reader.read_i32()

    await relationships.create(session["account_id"], target_id, "friend")
```

File: server/packet_handlers.py [hunk 5]
```diff
@@ -188,6 +192,59 @@ async def request_status_update_handler(session: "Session", packet_data: bytes):
...
+
+
+@bancho_handler(packets.ClientPackets.FRIEND_REMOVE)
+async def user_removes_friend_handler(session: "Session", packet_data: bytes):
+    packet_reader = packets.PacketReader(packet_data)
+    user_being_unfriended_id = packet_reader.read_i32()
```
Comment (cmyui) [inline server/packet_handlers.py]:
```suggestion
    target_id= packet_reader.read_i32()
```

File: server/packet_handlers.py [hunk 6]
```diff
@@ -188,6 +192,59 @@ async def request_status_update_handler(session: "Session", packet_data: bytes):
...
+
+    await relationships.remove(session["account_id"], user_being_unfriended_id)
```
Comment (cmyui) [inline server/packet_handlers.py]:
```suggestion
    await relationships.remove(session["account_id"], target_id)
```

File: server/repositories/channels.py [hunk 7]
```diff
@@ -23,7 +23,7 @@ async def create(
 ) -> dict[str, Any]:
     channel = await clients.database.fetch_one(
         query=f"""\
-            INSERT INTO channel (name, topic, read_privileges, write_privileges, auto_join, created_at, updated_at)
+            INSERT INTO channels (name, topic, read_privileges, write_privileges, auto_join, created_at, updated_at)
```
Comment (cmyui) [inline server/repositories/channels.py:26]: nice find & fix

**Review by James-Tepper — APPROVED** [review]

**Pull request merged**